pkg/kv/fs.go

```diff
@@ -0,0 +1,67 @@
+package kv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"sync"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type FSStore struct {
+	lock   *sync.RWMutex
+	fsPath string
+}
+
+func NewFSStore(logger *zap.Logger, fsPath string) *FSStore {
+	return &FSStore{
+		lock:   new(sync.RWMutex),
+		fsPath: fsPath,
+	}
+}
+
+func (s *FSStore) Start(ctx context.Context, g *errgroup.Group) error {
+	return nil
+}
+func (s *FSStore) Get(ctx context.Context, ns Namespace, key string) (string, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	path := fmt.Sprintf("%s/%s/%s", s.fsPath, ns, key)
+	b, err := os.ReadFile(path)
+
+	if errors.Is(err, os.ErrNotExist) {
+		s.Touch(path)
+		return "", nil
+	}
+	if err != nil {
+		return "", err
+	}
+	return string(b), nil
+}
+
+func (s *FSStore) Set(ctx context.Context, ns Namespace, key string, value string) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	path := fmt.Sprintf("%s/%s/%s", s.fsPath, ns, key)
+	s.Touch(path)
```

Comment:
According to https://pkg.go.dev/os#WriteFile, it will automatically create if necessary. You don't need to touch it.

Reply:
In my testing it will consistently throw `no such file or directory` error if not Touch the path beforehand.
I think it might be related to [this SO question about os.Create](https://stackoverflow.com/a/60500505/26586252), excerpt:

> ... you need to **ensure that the base directory exists**, because os.Create does not handle it.